syft/frameworks/torch/tensors/interpreters/native.py

```diff
@@ -856,6 +856,12 @@ def share(
             requires_grad (bool): Should we add AutogradTensor to allow gradient computation,
                 default is False.
         """
+        if crypto_provider == None:
+            raise ValueError ("'crypto_provider' should not be None.")
+        else:
+            if crypto_provider.is_client_worker:
```

Comment:
Use `elif`  here

You can be more explicit:
> ("'crypto_provider' should not be None.")
>  ("'crypto_provider' should not be None. Please provide a dedicated worker, which will act as a trusted third party and will provide the crypto primitives needed for MPC.")

Reply:
Done.

Reply:
Should I also place a check for a string incase someone provides a string rather than a worker to the crypto_provider so that it raises appropriate message of what type it should be?